test/test_make_equidistant.py

```diff
@@ -1,113 +1,179 @@
+    assume(not interp_inputs["t"][0] == interp_inputs["t"][-1])
+    # Ensure the timestamps are not in ascending order.
+    assume(not np.all(interp_inputs["t"][1:] >= interp_inputs["t"][:-1]))
+    # Check erroneous calls with descending timestamps.
+    with raises(ValueError):
+        # Reverse order of t and call make_equidistant().
+        make_equidistant(**interp_inputs)
 
-def test_t_new_to_dt_make_equidistant():
-    from numpy.testing import assert_almost_equal
 
-    t_new = make_equidistant(t, y, uy, dt)[0]
+@given(timestamps_values_uncertainties_kind())
+def test_t_new_to_dt_make_equidistant(interp_inputs):
+    # Ensure at least two different timestamps in the series.
+    assume(not interp_inputs["t"][0] == interp_inputs["t"][-1])
+    t_new = make_equidistant(**interp_inputs)[0]
     delta_t_new = np.diff(t_new)
     # Check if the new timestamps are ascending.
-    assert np.all(delta_t_new > 0)
-    # Check if the timesteps have the desired length.
-    assert_almost_equal(delta_t_new - dt, 0)
+    assert not np.any(delta_t_new < 0)
 
 
-def test_prev_in_make_equidistant():
-    kinds = ['previous', 'next', 'nearest']
-    for i_kind in kinds:
-        y_new, uy_new = make_equidistant(t, y, uy, dt, i_kind)[1:3]
-        # Check if all 'interpolated' values are present in the actual values.
-        assert np.all(np.isin(y_new, y))
-        assert np.all(np.isin(uy_new, uy))
+@given(timestamps_values_uncertainties_kind(kind_tuple=("previous", "next", "nearest")))
+def test_prev_in_make_equidistant(interp_inputs):
+    # Ensure at least two different timestamps in the series.
+    assume(not interp_inputs["t"][0] == interp_inputs["t"][-1])
+    y_new, uy_new = make_equidistant(**interp_inputs)[1:3]
+    # Check if all 'interpolated' values are present in the actual values.
+    assert np.all(np.isin(y_new, interp_inputs["y"]))
+    assert np.all(np.isin(uy_new, interp_inputs["uy"]))
 
 
-def test_linear_in_make_equidistant():
-    y_new, uy_new = make_equidistant(t, y, uy, dt, 'linear')[1:3]
-    # Check if all interpolated values lie in the range of the actual values.
-    assert np.all(np.amin(y) <= y_new)
-    assert np.all(np.amax(y) >= y_new)
-    # Check if for any non-zero input uncertainty the output contains non-zeros.
-    if np.any(uy):
-        assert np.any(uy_new)
+@given(timestamps_values_uncertainties_kind(kind_tuple=["linear"]))
+def test_linear_in_make_equidistant(interp_inputs):
+    # Ensure at least two different timestamps in the series.
+    assume(not interp_inputs["t"][0] == interp_inputs["t"][-1])
+    y_new, uy_new = make_equidistant(**interp_inputs)[1:3]
+    # Check if all interpolated values lie in the range of the original values.
+    assert np.all(np.amin(interp_inputs["y"]) <= y_new)
+    assert np.all(np.amax(interp_inputs["y"]) >= y_new)
 
 
 def test_linear_uy_in_make_equidistant():
     # Check for given input, if interpolated uncertainties equal 1 and
```

Comment:
is this test still necessary?

Reply:
Yes it is, to check at least against one expected result regarding the actual interpolation. Thanks for that!